src/sagas/index.js

```diff
@@ -191,33 +190,21 @@ function* fetchConsoles ({ payload: { vm } }) {
   const vmId = vm.get('id')
   const apiConsoles = yield callExternalAction('consoles', Api.consoles, { payload: { vmId } })
 
-  // headless VM (no graphics_consoles defined) - API response is `{}`
-  if (Object.values(apiConsoles).length === 0) {
-    yield put(setVmConsoles({ vmId, consolesList: [], selectedConsole: EMPTY_CONSOLES_LIST }))
-  }
-
-  if (apiConsoles && apiConsoles.graphics_console) {
-    const defaultConsoleProtocol = yield select(state => state.config.get('defaultConsole'))
-    const consoles = Transforms.VmConsoles.toInternal({ consoles: apiConsoles })
-    // select the console to use based on availability and app/user configs
-    const selectedConsole =
-      consoles.length === 0 ? EMPTY_CONSOLES_LIST
-        : consoles.length === 1 ? consoles[0]
-          : consoles.find(c => c.protocol === defaultConsoleProtocol) || NO_DEFAULT_CONSOLE
-
-    yield put(setVmConsoles({ vmId, consolesList: consoles, selectedConsole }))
-  }
+  // for headless VM (no graphics_consoles defined) - API response is `{}`
```

Comment:
Is this saga, `fetchConsoles`, necessary anymore?

Reply:
we can remove it